src/main/java/com/firebolt/jdbc/resultset/column/ColumnType.java

```diff
@@ -86,9 +98,9 @@ private static boolean isType(FireboltDataType fireboltDataType, String typeWith
 	private static List<ColumnType> getCollectionSubType(FireboltDataType fireboltDataType, String typeWithoutNullKeyword) {
 		String[] types;
 		for (String type: fireboltDataType.getAliases()) {
-			if(typeWithoutNullKeyword.startsWith(type.toUpperCase())) {
-				typeWithoutNullKeyword = RegExUtils.replaceFirst(typeWithoutNullKeyword, type.toUpperCase() + "\\(", "");
-				typeWithoutNullKeyword = StringUtils.substring(typeWithoutNullKeyword, 0, typeWithoutNullKeyword.length() - 1);
+			String typeUpperCase = type.toUpperCase();
+			if (typeWithoutNullKeyword.startsWith(typeUpperCase)) {
+				typeWithoutNullKeyword = typeWithoutNullKeyword.substring((typeUpperCase + "\\(").length() - 1, typeWithoutNullKeyword.length() - 1);
```

Comment:
What kind of utility function? I do not understand, sorry.

Reply:
Now, this is not something I feel strongly about, so up to you. However, in several cases we're stripping TYPE( ) from TYPE(INNER TYPE), like in the highlighted line - `typeWithoutNullKeyword.substring((typeUpperCase + "\\(").length() - 1, typeWithoutNullKeyword.length() - 1)`
To minimise duplication, especially since these lines are prone to "off by one" error, we can have a function like `getInnerType(string) -> string`.

Reply:
The purpose of this PR is not to solve all problems and duplications in this project but to remove one relatively heavy weighted dependency. I'd suggest to do other changes in separate commits.

Reply:
It's the problem introduced in this PR as a part of your change that makes it harder to read the code overall. The feedback here is purely based on this PR.
I'd rather have readable code than code that's 300kb less, especially since there were no issues with the JAR reported so far.